parts/k8s/kubernetesparams.t

```diff
@@ -599,3 +599,15 @@
       "type": "string"
     }
  {{end}}
+    ,"appGwSubnet": {
```

Comment:
Add a `"defaultValue": ""` property (see params definitions above) to fix the ARM errors. What's happening is that the ARM parameters spec is not being fulfilled by the template in cluster scenarios that don't assign values to these params (`pkg/engine/params_k8s.go`).

Reply:
instead of adding a default value, you can conditionally define those variables (see IsAzureCNI case above). This will avoid defining extra parameters when they are unused.